### PR TITLE
Bug 1167970 - Provide keyboard shortcuts for common actions

### PR DESCRIPTION
**Pull request by jacobwhite**

### CODE REVIEW

File: Client/Frontend/Strings.swift [hunk 1]
```diff
@@ -46,3 +46,16 @@ extension Strings {
...
+    public static let NewTabTitle = NSLocalizedString("Hotkeys.NewTab.DiscoveryTitle", value: "New Tab", comment: "Label to display in the Discoverability overlay for keyboard shortcuts")
+    public static let NewPrivateTabTitle = NSLocalizedString("Hotkeys.NewPrivateTab.DiscoveryTitle", value: "New Private Tab", comment: "Label to display in the Discoverability overlay for keyboard shortcuts")
+    public static let CloseTabTitle = NSLocalizedString("Hotkeys.CloseTab.DiscoveryTitle", value: "Close Tab", comment: "Label to display in the Discoverability overlay for keyboard shortcuts")
+    public static let ShowNextTabTitle = NSLocalizedString("Hotkeys.ShowNextTab.DiscoveryTitle", value: "Show Next Tab", comment: "Label to display in the Discoverability overlay for keyboard shortcuts")
+    public static let ShowPreviousTabTitle = NSLocalizedString("Hotkeys.ShowPreviousTab.DiscoveryTitle", value: "Show Prebious Tab", comment: "Label to display in the Discoverability overlay for keyboard shortcuts")
```
Comment (st3fan) [inline Client/Frontend/Strings.swift]: Typo in `Show Prebious Tab`.

File: Client/Frontend/Browser/BrowserViewController.swift [hunk 2]
```diff
@@ -1137,6 +1137,95 @@ class BrowserViewController: UIViewController {
...
+                tabManager.selectTab(tabManager.tabs[tabManager.count-1])
+            }
+        }
+    }
+
+    override var keyCommands: [UIKeyCommand]? {
+        if #available(iOS 9.0, *) {
+            return [
+                UIKeyCommand(input: "r", modifierFlags: .Command, action: #selector(BrowserViewController.reloadTab), discoverabilityTitle: Strings.ReloadPageTitle),
+                UIKeyCommand(input: "[", modifierFlags: .Command, action: #selector(BrowserViewController.goBack), discoverabilityTitle: Strings.BackTitle),
```
Comment (st3fan) [inline Client/Frontend/Browser/BrowserViewController.swift]: Two notes: we need both a `Back` and a `Forward` item. But these cannot be added statically. You have to take a look at the current Tab's and ask it if it is actually possible to go back or forward.

File: Client/Frontend/Browser/BrowserViewController.swift [hunk 3]
```diff
@@ -1137,6 +1137,95 @@ class BrowserViewController: UIViewController {
...
+                UIKeyCommand(input: "l", modifierFlags: .Command, action: #selector(BrowserViewController.openLocation), discoverabilityTitle: Strings.OpenLocationTitle),
+                UIKeyCommand(input: "t", modifierFlags: .Command, action: #selector(BrowserViewController.newTab), discoverabilityTitle: Strings.NewTabTitle),
+                UIKeyCommand(input: "p", modifierFlags: [.Command, .Shift], action: #selector(BrowserViewController.newPrivateTab), discoverabilityTitle: Strings.NewPrivateTabTitle),
+                UIKeyCommand(input: "w", modifierFlags: .Command, action: #selector(BrowserViewController.closeTab), discoverabilityTitle: Strings.CloseTabTitle),
+                UIKeyCommand(input: "\t", modifierFlags: .Control, action: #selector(BrowserViewController.nextTab), discoverabilityTitle: Strings.ShowNextTabTitle),
+                UIKeyCommand(input: "\t", modifierFlags: [.Control, .Shift], action: #selector(BrowserViewController.previousTab), discoverabilityTitle: Strings.ShowPreviousTabTitle),
+            ]
+        } else {
```
Comment (st3fan) [inline Client/Frontend/Browser/BrowserViewController.swift]: I am going to suggest that we drop iOS8 support. We will drop it globally from the app soon. Don't invest in it.

File: Client/Frontend/Browser/BrowserViewController.swift (hunk 2 above)
Comment (fluffyemily) [inline Client/Frontend/Browser/BrowserViewController.swift]: Yup, I was going to add that. Please add `]` as goForward.

File: Client/Frontend/Strings.swift [hunk 4]
```diff
@@ -46,3 +46,16 @@ extension Strings {
     public static let LoginsHelperDontSaveButtonTitle = NSLocalizedString("LoginsHelper.DontSave.Button", value: "Don’t Save", comment: "Button to not save the user's password")
     public static let LoginsHelperUpdateButtonTitle = NSLocalizedString("LoginsHelper.Update.Button", value: "Update", comment: "Button to update the user's password")
 }
+
+//Hotkey Titles
+extension Strings {
+    public static let ReloadPageTitle = NSLocalizedString("Hotkeys.Reload.DiscoveryTitle", value: "Reload Page", comment: "Label to display in the Discoverability overlay for keyboard shortcuts")
+    public static let BackTitle = NSLocalizedString("Hotkeys.Back.DiscoveryTitle", value: "Back", comment: "Label to display in the Discoverability overlay for keyboard shortcuts")
+    public static let FindTitle = NSLocalizedString("Hotkeys.Find.DiscoveryTitle", value: "Find", comment: "Label to display in the Discoverability overlay for keyboard shortcuts")
+    public static let OpenLocationTitle = NSLocalizedString("Hotkeys.OpenLocation.DiscoveryTitle", value: "Open Location", comment: "Label to display in the Discoverability overlay for keyboard shortcuts")
```
Comment (fluffyemily) [inline Client/Frontend/Strings.swift]: I'm not sure if `Open Location` is the right wording here. I'd suggest something more like `Search or Enter Address` which is the placeholder for the Location Text Field.

File: Client/Frontend/Browser/BrowserViewController.swift (hunk 3 above)
Comment (jacobwhite) [inline Client/Frontend/Browser/BrowserViewController.swift]: How soon?

File: Client/Frontend/Strings.swift (hunk 4 above)
Comment (jacobwhite) [inline Client/Frontend/Strings.swift]: Firefox desktop uses "Select Location Bar"

**Pull request merged**